freedesktop.org.xml.in

```diff
@@ -6316,4 +6320,15 @@ command to generate the output files.
     <glob pattern="*.stl"/>
   </mime-type>
 
+<mime-type type="application/x-gcode">
+    <_comment>G-code file</_comment>
+    <sub-class-of type="text/plain"/>
+    <generic-icon name="text-x-generic"/>
+    <magic priority="10">
+      <match type="string" value="; " offset="0"/>
+      <match type="string" value="( " offset="0"/>
```

Comment:
> Aah. Never seen that in RepRap related GCODE.

Me neither, but I researched a bit to see if we could include more CNC machines and I came across that way of commenting.

I thought about adding `G` or `M` for files with no comments but there could be a lot plain-text-like files starting this way, however starting with `;` seems rather singular. Starting with a `(` might be pretty common too though.

> Also Cura Lulzbot does this (no space)

The space should be removed then.